src/api/firebase.js

```diff
@@ -179,21 +181,63 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 		dateCreated: new Date(),
 		// NOTE: This is null because the item has just been created.
 		// We'll use updateItem to put a Date here when the item is purchased!
-		dateLastPurchased: null,
+		dateLastPurchased: [new Date()],
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
+	console.log(newItem);
 	return newItem;
 }
 
+//Helper function to handle estimation logic
+async function handleCalculateEstimate(listRef) {
+	const selectedItem = await getDoc(listRef);
+
+	//access array of dateLastPurchased for the selected item
+	const selectedLastPurchase = selectedItem.data().dateLastPurchased;
+
+	//retrieve the most recent purchase date from the array and convert to a date
+	const lastPurchased =
+		selectedLastPurchase[selectedLastPurchase.length - 1].toDate();
+
+	//calculates the previously estimated days until next purchase
+	const previousEstimate = getDaysBetweenDates(
+		lastPurchased,
+		selectedItem.data().dateNextPurchased.toDate(),
+	);
+	let daysSincePrevPurchase = getDaysBetweenDates(lastPurchased, new Date());
+
+	//calculates a new estimate based on the previous estimate, days since last purchased, and total purchases
+	let newEstimate = calculateEstimate(
+		previousEstimate,
+		daysSincePrevPurchase,
+		selectedItem.data().totalPurchases,
+	);
+
+	return { newEstimate, selectedLastPurchase };
+}
+
 export async function updateItem(listPath, itemID, isChecked) {
 	const listRef = doc(db, listPath, 'items', itemID);
 
-	await updateDoc(listRef, {
-		dateLastPurchased: isChecked ? new Date() : null,
-		totalPurchases: isChecked ? increment(1) : increment(-1),
-	});
+	let { newEstimate, selectedLastPurchase } =
+		await handleCalculateEstimate(listRef);
+
+	//checks to see if checkbox is checked in front end, updates selected item
+	if (isChecked) {
+		await updateDoc(listRef, {
+			dateLastPurchased: [...selectedLastPurchase, new Date()],
+			dateNextPurchased: getFutureDate(newEstimate),
+			totalPurchases: increment(1),
+		});
+	} else {
```

Comment:
praise: when testing I also tried unchecking the cheesecake item to see what happens in Firebase and it looks great!